src/topology/algebra/group.lean

```diff
@@ -329,3 +330,60 @@ instance : topological_add_group α :=
   end⟩
 
 end add_group_with_zero_nhd
+
+section filter_mul
+local attribute [instance]
+  set.pointwise_one set.pointwise_mul set.pointwise_add filter.pointwise_mul filter.pointwise_add
+
+section
+variables [topological_space α] [group α] [topological_group α]
+
+@[to_additive is_open_pointwise_add_left]
+lemma is_open_pointwise_mul_left {s t : set α} : is_open t → is_open (s * t) := λ ht,
+begin
+  have : ∀a, is_open ((λ (x : α), a * x) '' t),
+    assume a, apply is_open_map_mul_left, exact ht,
+  rw pointwise_mul_eq_Union_mul_left,
+  exact is_open_Union (λa, is_open_Union $ λha, this _),
+end
+
+@[to_additive is_open_pointwise_add_right]
+lemma is_open_pointwise_mul_right {s t : set α} : is_open s → is_open (s * t) := λ hs,
+begin
+  have : ∀a, is_open ((λ (x : α), x * a) '' s),
+    assume a, apply is_open_map_mul_right, exact hs,
+  rw pointwise_mul_eq_Union_mul_right,
+  exact is_open_Union (λa, is_open_Union $ λha, this _),
+end
+
+end
+
+section
+variables [topological_space α] [comm_group α] [topological_group α]
+
+@[to_additive nhds_add_nhds]
+lemma nhds_mul_nhds (x y : α) : nhds x * nhds y = nhds (x * y) :=
```

Comment:
This exhibits `nhds` as a `is_mul_hom` and probably as `is_monoid_hom`.

Reply:
I can also imagine that the lemma is just as useful in the other direction. Which would be more conform the "hom" direction in mathlib.

Reply:
Looks like `is_monoid_hom` cannot be proved. That would require `nhds 1 = principal {1}`, which means `{1} ∈ nhds 1` and so `{1}` must be open.